src/autoscaler.ts

```diff
@@ -122,40 +122,60 @@ export default class AutoscaleProcessor {
         if (scaleMetrics && scaleMetrics.length > 0) {
             // check if we should scale up the group
             if (this.evalScaleConditionForAllPeriods(ctx, scaleMetrics, count, group, 'up')) {
-                desiredCount = desiredCount + group.scalingOptions.scaleUpQuantity;
-                if (desiredCount > group.scalingOptions.maxDesired) {
-                    desiredCount = group.scalingOptions.maxDesired;
-                }
+                if (await this.instanceGroupManager.allowAutoscalingByDirection(group.name, 'up')) {
```

Comment:
We want to be able to react fast especially to scaling up and this might alter this behaviour. 

- For example, let's say scaleUpPeriodsCount = 3, and scaleUpThreshold = 5. At time x, we might see the following metrics per period: 4, 3, 3. Which means we scale up. 

- We wait for gracePeriodTTLSec for the instances to come up and be operational and then we check again the new metrics window. But if the usage is high, the newly instances will be immediately used and the sliding window will become 3, 3, 2.  Which means we should scale up again.
- You change suggests we should not scale up then, but instead wait 2 more scalePeriod and only then take the decision. This might be too late. 

If we do want to play with having the grace period larger, I would prefer to leave the choice of the grace period in the hands of the admin of the group. Maybe we should look into adding a configurable grace period per direction per group, and the admin can set it to the desired value, such as scaleUpPeriodsCount * group.scalingOptions.scalePeriod for up.